### PR TITLE
feat: Add support for eventstream input & initial request to RPC-based protocols

### PR DESCRIPTION
**Pull request by sichanyoo**
## Issue \#

**Companion PR: https://github.com/awslabs/aws-sdk-swift/pull/1377**

**For RPC-based protocols:**

Eventstream input ticket: https://github.com/awslabs/aws-sdk-swift/issues/1175
Initial request ticket: https://github.com/awslabs/aws-sdk-swift/issues/1172

## Description of changes

- Add a new SwiftIntegration that generates `func makeInitialRequestMessage` in extension of input structs when: 1) protocol is RPC, 2) protocol supports event streaming, 3) upstream input shape has streaming union member.
- Hook up `EventStreamBodyMiddleware` codegen for RPC protocols that support event streaming.
- Make `EventStreamBodyMiddleware` take in `initialRequestMessage` in constructor. The middleware passes this value to `DefaultMessageEncoderStream` constructor.
- Make `DefaultMessageEncoderStream` return `initialRequestMessage` as the first item in stream if it is non-nil and was set during construction.
- Hook up Encodable conformance codegen for input shapes in RPC protocols if: 1) protocol supports event streaming & 2) input shape has streaming union member. Exclude streaming member in encoding, because the encoding functionality provided by this conformance is used to encode input struct into an initial request message without streaming member.

## Scope

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputBodyMiddleware.kt [hunk 1]
```diff
@@ -137,9 +148,12 @@ class OperationInputBodyMiddleware(
         )
     }
 
-    private fun addEventStreamMiddleware(writer: SwiftWriter, operationStackName: String, inputSymbol: Symbol, outputSymbol: Symbol, payloadSymbol: Symbol, keyPath: String, defaultBody: String) {
+    private fun addEventStreamMiddleware(writer: SwiftWriter, operationStackName: String, inputSymbol: Symbol, outputSymbol: Symbol, payloadSymbol: Symbol, keyPath: String, defaultBody: String, sendInitialRequest: Boolean) {
+        if (sendInitialRequest) {
+            writer.write("let initialRequestMessage = try input.makeInitialRequestMessage(encoder: encoder)")
```
Comment (jbelkins) [inline smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputBodyMiddleware.kt:153]: Does `makeInitialRequestMessage(encoder: encoder)` get generated somewhere?

File: Sources/ClientRuntime/EventStream/DefaultMessageEncoderStream.swift [hunk 2]
```diff
@@ -34,47 +37,56 @@ extension EventStream {
...
-                    return nil
+                    return try await processEventToData(event: event)
                 }
+            }
 
+            mutating func processEventToData(event: Event) async throws -> Data {
                 // marshall event to message
-                let message = try event.marshall(encoder: requestEncoder)
-
+                let  message = try event.marshall(encoder: requestEncoder)
```
Comment (jbelkins) [inline Sources/ClientRuntime/EventStream/DefaultMessageEncoderStream.swift:85]: nit: 2 spaces between `let` & `message`

File: smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputBodyMiddleware.kt (hunk 1 above)
Comment (sichanyoo) [reply smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/OperationInputBodyMiddleware.kt:153]: Moved the codegen for this method from aws-sdk-swift to smithy-swift as pointed out in offline discussion, given initial request is part of the generic Smithy specification.

File: Sources/ClientRuntime/EventStream/DefaultMessageEncoderStream.swift (hunk 2 above)
Comment (sichanyoo) [reply Sources/ClientRuntime/EventStream/DefaultMessageEncoderStream.swift]: Fixed.

**Review by jbelkins — APPROVED** [review]

**Pull request merged**